pkg/abaputils/manageGitRepositoryUtils.go

```diff
@@ -80,6 +81,37 @@ func PrintLogs(entity PullEntity) {
 	log.Entry().Info("-------------------------")
 }
 
+//GetRepositories for parsing  one or multiple branches and repositories from repositories file or branchName and repositoryName configuration
+func GetRepositories(config *RepositoriesConfig) ([]Repository, error) {
+	var repositories = make([]Repository, 0)
+	if reflect.DeepEqual(RepositoriesConfig{}, config) {
+		return repositories, fmt.Errorf("Failed to read repository configuration: %w", errors.New("Eror in configuration, most likely you have entered empty or wrong configuration values. Please make sure that you have correctly specified the branches in the repositories to be checked out"))
+	}
+	if config.RepositoryName == "" && config.BranchName == "" && config.Repositories == "" && len(config.RepositoryNames) == 0 {
+		return repositories, fmt.Errorf("Failed to read repository configuration: %w", errors.New("You have not specified any repository configuration to be pulled into the ABAP Environment System. Please make sure that you specified the repositories with their branches that should be pulled either in a dedicated file or via in-line configuration. For more information please read the User documentation"))
+	}
+	if config.Repositories != "" {
+		descriptor, err := ReadAddonDescriptor(config.Repositories)
+		if err != nil {
+			return repositories, err
+		}
+		err = CheckAddonDescriptorForRepositories(descriptor)
+		if err != nil {
+			return repositories, fmt.Errorf("Could not parse config file %v, %w", config.Repositories, err)
```

Comment:
I think this is misleading. The file was successfully parsed in "ReadAddonDescriptor".
"CheckAddonDescriptorForRepositories" would e.g. fail if no repositories were specified.

E.g. the customer creates an addon.yml for the specifics of an Addon Product and decides to pass the repos via "repositoryNames". He is missing the repositories in the yml file - but this is not a parsing issue.

Reply:
Ok i know what you mean. Just for further clarification I changed it to 'Error in config file'. That should solve it.

Reply:
Message would now be e.g. "Error in config file repositories.yml, AddonDescriptor doesn't contain any repositories"